cvmfs/sync_mediator.cc

```diff
@@ -775,47 +784,98 @@ void SyncMediator::CreateNestedCatalog(SharedPtr<SyncItem> directory) {
 
 void SyncMediator::RemoveNestedCatalog(SharedPtr<SyncItem> directory) {
   const std::string notice =  "Nested catalog at " + directory->GetUnionPath();
-  PrintChangesetNotice(kRemoveCatalog, notice);
+  reporter_->OnRemove(notice, catalog::DirectoryEntry());
 
   if (!params_->dry_run) {
     catalog_manager_->RemoveNestedCatalog(directory->GetRelativePath());
   }
 }
 
+void SyncDiffReporter::OnInit(const history::History::Tag &from_tag,
```

Comment:
Since we are not using the parameters, let's comment their names like

    const history::History::Tag & /* from_tag */

This should also avoid warning with some compilers.